sima/sequence.py

```diff
@@ -1148,6 +1148,55 @@ def _todict(self, savedir=None):
         }
 
 
+class _SplicedSequence(_WrapperSequence):
+    """Sequence for splicing together nonconsecutive frames
+
+    Parameters
+    ----------
+    base : Sequence
+
+    times : list of frame indices of the base sequence
+
+    """
+    def __init__(self, base, times):
+        super(_SplicedSequence, self).__init__(base)
+        self._base_len = len(base)
+        self._times = times
+
+    def __iter__(self):
+        try:
+            for t in self._times:
+                # Not sure if np.copy is needed here (see _IndexedSequence)
+                yield np.copy(self._base._get_frame(t))
+        except NotImplementedError:
+            if self._indices[0].step < 0:
```

Comment:
Not sure where the `self._indices` comes from.
What about `if np.any(np.diff(self._times) <= 0):` ?

Reply:
I *think this was copied in error over from indexed sequence. I can clean up the logic and then resubmit.